plugins/outputs/file/file.go

```diff
@@ -18,19 +19,47 @@ import (
 //go:embed sample.conf
 var sampleConfig string
 
+var ValidCompressionAlgorithmLevels = map[string][]int{
+	"zstd":     {1, 3, 7, 11},
+	"gzip":     {-2, -1, 1, 9},
+	"zlib":     {-2, -1, 1, 9},
+	"identity": {0},
+}
+
 type File struct {
-	Files               []string        `toml:"files"`
-	RotationInterval    config.Duration `toml:"rotation_interval"`
-	RotationMaxSize     config.Size     `toml:"rotation_max_size"`
-	RotationMaxArchives int             `toml:"rotation_max_archives"`
-	UseBatchFormat      bool            `toml:"use_batch_format"`
-	Log                 telegraf.Logger `toml:"-"`
+	Files                []string        `toml:"files"`
+	RotationInterval     config.Duration `toml:"rotation_interval"`
+	RotationMaxSize      config.Size     `toml:"rotation_max_size"`
+	RotationMaxArchives  int             `toml:"rotation_max_archives"`
+	UseBatchFormat       bool            `toml:"use_batch_format"`
+	CompressionAlgorithm string          `toml:"compression_algorithm"`
+	CompressionLevel     int             `toml:"compression_level"`
+	encoder              internal.ContentEncoder
+	Log                  telegraf.Logger `toml:"-"`
 
 	writer     io.Writer
 	closers    []io.Closer
 	serializer serializers.Serializer
 }
 
+func validateCompressionAlgorithm(algorithm string) error {
+	for validAlgorithm := range ValidCompressionAlgorithmLevels {
+		if algorithm == validAlgorithm {
+			return nil
+		}
+	}
+	return fmt.Errorf("unknown or unsupported algorithm provided: %s", algorithm)
+}
+
+func validateCompressionLevel(algorithm string, level int) error {
+	for _, validAlgorithmLevel := range ValidCompressionAlgorithmLevels[algorithm] {
+		if level == validAlgorithmLevel {
+			return nil
+		}
+	}
```

Comment:
This should be done using [`internal.NewContentEncoder`](https://github.com/influxdata/telegraf/blob/09adec7395366698bd20225b29eb807fe042037b/internal/content_coding.go#L91). This way it is easily extensible to more compression algorithms as soon as they are added.